NDTensors/ext/NDTensorsMetalExt/typeparameteraccessors.jl

```diff
@@ -0,0 +1,20 @@
+# # `TypeParameterAccessors.jl` overloads.
+using NDTensors.TypeParameterAccessors:
+  TypeParameterAccessors, Position, parameter, position, set_parameter
+## TODO this seems like a `GPUArrays` generic function
+storagemode(T::Type{<:MtlArray}) = parameter(T, 3)
+## TODO this seems like a `GPUArrays` generic function
+TypeParameterAccessors.position(::Type{<:MtlArray}, ::typeof(storagemode)) = Position(3)
+
+function TypeParameterAccessors.default_parameter(::Type{<:MtlArray}, ::typeof{eltype})
+  return Float32
+end
+TypeParameterAccessors.default_parameter(::Type{<:MtlArray}, ::typeof{ndims}) = 1
```

Comment:
Seems like these could be generic `AbstractArray` definitions, besides the element type.

There seems to be some subtlety here, for `CuArray` we make the eltype default to `Float64` while for `MtlArray` we make the eltype default to `Float32`.

I understand the reason for Metal is because `Float64` isn't supported, but for CUDA I could imagine also making it `Float32` since that is faster in general.

Another perspective is that we just make it `Float64` for any `AbstractArray`, just to keep a simple rule, and for Metal users have to specify `Float32` (I'm leaning towards that approach but I don't know what the practical implications of that are).

Reply:
My thought was just that if you did something like `ITensor(MtlArray, (Index(3), Index(4)))` you would want the code to work without throwing an error that Metal doesn't support Float32.  
The de-facto "default" in CUDA.jl is `Float32` so I could see us having `AbstractGPUArray`s in general have a default of `Float32`. Right now I have the implementation of all `AbstractArrays` are `Float64` and we can change that later relatively easily if we decide otherwise

Reply:
That's a good point, since that was the main motivation for the default parameter system. I think it makes sense to make `Float64` the default for `AbstractArray`, and `Float32` the default for `AbstractGPUArray`.